.github/actions/setup-python-poetry/action.yaml

```diff
@@ -0,0 +1,47 @@
+name: "Install Python and Poetry"
+description: "Add Poetry, dependency manager for Python"
+
+inputs:
+  python-version: # Dynamically set Python version
```

Comment:
Yes, the github action naming convention is kebab case. If I changed this to be `snake_case`, it would clash will all the other github actions out there (and then we start having to do a mental guessing game: is it `snake_case` or is it `kebab-case`? Going to stick with `kebab-case` for now.